lib/index.js

```diff
@@ -62,6 +66,21 @@ function start(startingConfig) {
   };
 }
 
+function setFlushOnExit() {
+  function handleBeforeExit(exitCode) {
+    debug('flushing last beacons before exiting');
+    transmitter.handlePeriodicTasks()
+      .then(() => {
+        process.exit(exitCode);
```

Comment:
In this case I think that exiting is dangerous. We run inside another app that maybe has its own listeners on kill events.. We should think about this thoroughly

Reply:
that's why I'm documenting this feature and allowing it to be turned off.
do you think it's not enough?

Reply:
check out the new approach, might be better